File: HackerRank/Python/Medium/M0004.py
# Problem Statement: https://www.hackerrank.com/challenges/iterables-and-iterators/problem

from itertools import combinations

_, N = int(input()), input().split()
K = int(input())
count = [1 if 'a' in comb else 0 for comb in combinations(N, K)]
print('%.3f' % (sum(count)/len(count)))

File: HackerRank/C/Medium/M0004.c
/*
Problem Statement: https://www.hackerrank.com/challenges/printing-tokens-/problem
*/

#include <stdio.h>
#include <string.h>

int main() {
	char s[1000];
	fgets(s, sizeof(s), stdin);
	char* token = strtok(s, " ");
	
	while (token) {
		printf("%s\n", token);
		token = strtok(NULL, " ");
	}

	return 0;
}